README.md
# Flurry SDK

[![pod](https://img.shields.io/cocoapods/v/Flurry-iOS-SDK)](https://cocoapods.org/pods/Flurry-iOS-SDK)
[![platform](https://img.shields.io/cocoapods/p/Flurry-iOS-SDK)](https://cocoapods.org/pods/Flurry-iOS-SDK)
[![license](https://img.shields.io/github/license/flurry/flurry-ios-sdk)](https://github.com/flurry/Flurry-iOS-SDK)

## Table of Contents

- [Installation](#installation)
  - [iOS](#ios)
  - [watchOS](#watchos)
  - [tvOS](#tvos)
- [Requirements](#requirements)
- [Examples](#examples)
- [Suppport](#support)
- [License](#license)

## Installation

To install FlurrySDK from CocoaPods, please follow the instructions below. Remember to include `use_frameworks!` if your app target is in Swift.

### iOS

To enable Flurry Analytics:

```ruby
pod 'Flurry-iOS-SDK/FlurrySDK'
```

To enable Flurry Ad serving: 

```ruby
pod 'Flurry-iOS-SDK/FlurrySDK'
pod 'Flurry-iOS-SDK/FlurryAds'
```

To enable Flurry Config:

```ruby
pod 'Flurry-iOS-SDK/FlurryConfig'
```

To enable Flurry Messaging:

```ruby
pod 'Flurry-iOS-SDK/FlurryMessaging'
```

### watchOS

To use FlurrySDK for Apple Watch Extension:    

```ruby
target 'Your Apple Watch 2.x Extension Target' do 
  platform :watchos, '2.0'
  pod 'Flurry-iOS-SDK/FlurrySDK'
end   
```

### tvOS

To use FlurrySDK for tvOS apps:

```ruby
target 'Your tvOS Application' do
  platform :tvos, '10.0'
  pod 'Flurry-iOS-SDK/FlurrySDK'
end
```

To enable Flurry Messaging for tvOS:

```ruby
pod 'Flurry-iOS-SDK/FlurryMessaging'
```

## Requirements

* iOS 10.0+
* tvOS 10.0+
* watchOS 3.0+

## Examples

Listed below are brief examples of initializing and using Flurry in your project. For detailed instructions, please [check our documentation](https://developer.yahoo.com/flurry/docs/).

### Initialize Flurry

* iOS/tvOS

  To initialize Flurry, please import Flurry in your Application Delegate and start a Flurry session inside `applicationDidFinishLaunching`, as described below.

  ```swift
  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
      let sb = FlurrySessionBuilder()
            .build(logLevel: FlurryLogLevel.all)
            .build(crashReportingEnabled: true)
            .build(appVersion: "1.0")
            .build(iapReportingEnabled: true)
        
      Flurry.startSession(apiKey: "YOUR_API_KEY", sessionBuilder: sb)
      return true
  }
  ```

* watchOS

  Please follow [our instructions here](https://developer.yahoo.com/flurry/docs/integrateflurry/watchos/).

### Log Events

Use this to log normal events and timed events in your app.

* iOS/tvOS

  ```swift
  // Normal events
  Flurry.log(eventName: "Event", parameters: ["Key": "Value"])
        
  // Timed events
  Flurry.log(eventName: "Event", parameters: ["Key": "Value"], timed: true)
  Flurry.endTimedEvent(eventName: "Event", parameters: ["Key": "Value"])
        
  // Standard events
  let param = FlurryParamBuilder()
      .set(doubleVal: 34.99, param: FlurryParamBuilder.totalAmount())
      .set(booleanVal: true, param: FlurryParamBuilder.success())
      .set(stringVal: "book 1", param: FlurryParamBuilder.itemName())
      .set(stringVal: "This is an awesome book to purchase !!!", key: "note")
        
  Flurry.log(standardEvent: FlurryEvent.purchased, param: param)
  ```
  Please see our [sample project here](https://github.com/flurry/iOS-StandardEventSample).


* watchOS

  ```swift
  FlurryWatch.logWatchEvent("Event", withParameters: ["Key": "Value"])
  ```

### Log Error (iOS/tvOS)

Use this to log exceptions and/or errors that occur in your app. Flurry will report the first 10 errors that occur in each session.

```swift
Flurry.log(errorId: "ERROR_NAME", message: "ERROR_MESSAGE", exception: e)
```

### Track User Demographics (iOS/tvOS)

After identifying the user, use this to log the user’s assigned ID, username, age and gender in your system.

```swift
Flurry.set(userId: "USER_ID")
Flurry.set(age: 20)
Flurry.set(gender: "f") // "f" for female and "m" for male
```

### Session Origins and Attributes (iOS/tvOS)

This allows you to specify session origin and deep link attached to each session, or add a custom parameterized session parameters. You can also add an SDK origin specified by origin name and origin version.

```swift
Flurry.add(sessionOriginName: "SESSION_ORIGIN")
Flurry.add(sessionOriginName: "SESSION_ORIGIN", deepLink: "DEEP_LINK")
Flurry.sessionProperties(["key": "value"])
Flurry.add(originName: "ORIGIN_NAME", originVersion: "ORIGIN_VERSION")
Flurry.add(originName: "ORIGIN_NAME", originVersion: "ORIGIN_VERSION", parameters: ["key": "value"])
```

## Support

* [Flurry Documentation](https://developer.yahoo.com/flurry/docs/)
* [FAQs for Flurry](https://developer.yahoo.com/flurry/docs/faq/)
* [Flurry Support](https://developer.yahoo.com/support/flurry/)

## License

Copyright (c) 2021 Yahoo. All rights reserved.

This project is licensed under the terms of the [Apache 2.0](http://www.apache.org/licenses/LICENSE-2.0) open source license. Please refer to [LICENSE](LICENSE) for the full terms. Your use of Flurry is governed by [Flurry Terms of Service](https://developer.yahoo.com/flurry/legal-privacy/terms-service/).


CHANGELOG.md
# Change Log

## Version 12.x

### Version 12.4.0 - 08/16/2023

* Resolves crash with timedId when string is very long
* Minimum iOS version is 12.0

### Version 12.3.1 - 06/16/2023

* Adds new apis to provide more flexibility for closing timed events

### Version 12.2.0 - 01/24/2023

* Version is built on Xcode 14
* New api added to FlurrySessionBuilder to simplify gpp consent implementation
* Watch support is temporarily removed from the podspec due to the following issue: https://github.com/CocoaPods/CocoaPods/issues/11558
* Use manual installation or Swift Package for WatchOS support

### Version 12.1.4 - 11/02/2022

* Revert back to 12.1.1 due to conflict with ads sdk

### Version 12.1.3 - 10/27/2022

* Version is built on Xcode 14
* Watch support is temporarily removed from the podspec due to the following issue: https://github.com/CocoaPods/CocoaPods/issues/11558
* Use manual installation or Swift Package for WatchOS support

### Version 12.1.2 - 

* internal release only


### Version 12.1.1 - 06/30/2022

* Fix for http 400 error when consentStrings is nil

### Version 12.1.0 - 06/23/2022

* Flurry's xcframework uses framework rather than static libraries
* Fix for tvOS arm64 simulator support 


### Version 12.0.0 - 04/18/2022

* Introduced the XCFramework in our build pipeline to replace legacy static library 
* Enhanced the minimum deployment target to iOS 10.0, tvOS 10.0, macOS 11.0, and watchOS 3.0
* Introduced `NS_SWIFT_NAME()` in header file for re-defining all external Flurry APIs and enums in swift
* APIs removed :
    * `[Flurry setSessionReportsOnCloseEnabled:]`
    * `[Flurry setSessionReportsOnPauseEnabled:]`
    * `[Flurry logPaymentTransactionParamsWithTransactionId:productId:currency:productName:transactionState:userDefinedParams:statusCallback:]`
    * `[Flurry logFlurryPaymentTransactionParamsWithTransactionId:productId:quantity:price:currency:productName:transactionState:userDefinedParams:statusCallback:]`
* Fixed the background session reporting issue that caused long session duration
* Fixed `FlurryConsent` crashes
    
    
## Version 11.x

### Version 11.4.0 - 11/10/2021

* Introduced drop rule for user property
* Added new APIs for `Flurry`
    * Add new manual payment log API `[Flurry logPaymentTransactionWithTransactionId:productId:quantity:price:currency:productName:transactionState:userDefinedParams:statusCallback:]`
    * Add new setter APIs
        * Add `[Flurry setAppVersion:]`
        * Add `[Flurry setSessionContinueSeconds:]`
        * Add `[Flurry setCountBackgroundSessions:]`
* API deprecations
    * `[Flurry logFlurryPaymentTransactionParamsWithTransactionId:productId:quantity:price:currency:productName:transactionState:userDefinedParams:statusCallback:]`
    
* Fixed publisher segmentation registering internal listener during initialization
* Fixed FlurryConfig memory leaks during initialization
* Update Copyright (c) 2021 Yahoo

    
### Version 11.3.0 - 08/12/2021

* You asked for standard events and we are here to tell you, you are getting them! Standard Events help you track standardized actions that users take within your app – for example, making a purchase, adding a social comment, or clicking on an Ad. We are taking the guess work out on what to call the events and what parameters to pass in by providing an API to seamlessly log all of these standard actions within your app. 

* Fix missing crash report on tvOS SDK

### Version 11.2.1 - 04/06/2021

* Separates xcode11 and xcode 12 builds
* FlurryMessaging auto integration crash fix
* Fix UIApplication.endBackgroundTask(_:) risk
* Fix incorrect crash report stacktrace for log error and exception

### Version 11.2.0 - 11/30/2020

* Separates xcode11 and xcode 12 builds
* Symbolication error fix
* Background sessions fix
* App orientation crash fix

### Version 11.1.1 - 10/08/2020

* Fixed location source threading warning
* Potentially fixed interface orientation data source crash

### Version 11.1.0 - 09/15/2020

* Fixed duplicate symbol issue in crash reporter
* Upgraded APIs level for Interface orientation data source and fixed potential race condition

### Version 11.0.0 - 08/24/2020

* Flurry Analytics Support for iOS 14.0, tvOS 14.0, watchOS 7.0
* Added support for iOS 14 App Tracking Transparency. [See YDN doc here](https://developer.yahoo.com/flurry/docs/analytics/gettingstarted/ios14/att/)
* Added SKAdNetwork Conversion Value: Publishers can use Flurry to manage & understand Conversion Value. [See Blog Post](https://www.flurry.com/blog/post/626449503277531136/announcing-flurry-conversion-value-analytics-for) and [YDN doc](https://developer.yahoo.com/flurry/docs/analytics/gettingstarted/ios14/SKAdNetwork/) here.
* New APIs for `FlurrySKAdNetwork`
    * Added `[FlurrySKAdNetwork flurryUpdateConversionValueWithEvent:]`
    * Added `[FlurrySKAdNetwork flurryUpdateConversionValue:]`
* New APIs for `Flurry`
    * `[Flurry logPaymentTransactionParamsWithTransactionId:productId:quantity:price:currency:productName:transactionState:userDefinedParams:statusCallback:]`
* API Deprecations
    * `[Flurry logPaymentTransactionParamsWithTransactionId:productId:quantity:price:currency:productName:transactionState:userDefinedParams:statusCallback:]` 
* API Removals
    * `[Flurry logAllPageViewsForTarget:]`
    * `[Flurry stopLogPageViewsForTarget:]`
    * `[Flurry logPageView]`
* Fixed bugs in FlurryKVStorage for nested data types
* Fixed crash on FlurrySignalConnectionTypeDataProvider KVC issue


## Version 10.x

### Version 10.3.4 - 07/23/2020

* Fixed crash analytics issue
* Fixed App orientation data validation crash
* Removed base64 encoded params for error frame

### Version 10.3.3 - 06/24/2020

* Fixed unarchiving exception with NSSecureCoding conformed
* Fixed watchOS / iOS SDK connectivity issues

### Version 10.3.2 - 06/11/2020

* Fixed memory leaks in FlurryDataSender and FlurryDataSenderIndex
* Fixed crash in FlurryiOSUtil while getting CoreTelephony data

navtreeindex0.js
var NAVTREEINDEX0 =
{
"_f_config_8h_source.html":[2,0,0,0,2,0],
"_flurry_8h_source.html":[2,0,0,0,2,1],
"_flurry_ad_banner_8h_source.html":[2,0,0,0,0,0],
"_flurry_ad_banner_delegate_8h_source.html":[2,0,0,0,0,1],
"_flurry_ad_delegate_8h_source.html":[2,0,0,0,1,0],
"_flurry_ad_error_8h.html":[2,0,0,0,0,2],
"_flurry_ad_error_8h.html#a81481baf8f36bdf16f3b613dee6332db":[2,0,0,0,0,2,0],
"_flurry_ad_error_8h.html#a81481baf8f36bdf16f3b613dee6332dba0c2627398e3ba3b94478d746a3269c91":[2,0,0,0,0,2,0,1],
"_flurry_ad_error_8h.html#a81481baf8f36bdf16f3b613dee6332dba1bb6e553d9b03d112ad5674317b7c0a7":[2,0,0,0,0,2,0,2],
"_flurry_ad_error_8h.html#a81481baf8f36bdf16f3b613dee6332dba2e6d518cbef48fa474a0146d05cd47ca":[2,0,0,0,0,2,0,0],
"_flurry_ad_error_8h_source.html":[2,0,0,0,0,2],
"_flurry_ad_interstitial_8h_source.html":[2,0,0,0,0,3],
"_flurry_ad_interstitial_delegate_8h_source.html":[2,0,0,0,0,4],
"_flurry_ad_native_8h_source.html":[2,0,0,0,0,5],
"_flurry_ad_native_asset_8h_source.html":[2,0,0,0,0,6],
"_flurry_ad_native_delegate_8h_source.html":[2,0,0,0,0,7],
"_flurry_ad_native_style_8h_source.html":[2,0,0,0,0,8],
"_flurry_ad_stream_delegate_8h_source.html":[2,0,0,0,0,9],
"_flurry_ad_targeting_8h_source.html":[2,0,0,0,0,10],
"_flurry_ads_8h_source.html":[2,0,0,0,1,1],
"_flurry_native_ad_cell_8h_source.html":[2,0,0,0,0,11],
"_flurry_session_builder_8h_source.html":[2,0,0,0,2,2],
"_flurry_tumblr_8h_source.html":[2,0,0,0,3,0],
"_flurry_tumblr_delegate_8h_source.html":[2,0,0,0,3,1],
"_flurry_watch_8h_source.html":[2,0,0,0,2,3],
"annotated.html":[1,0],
"classes.html":[1,1],
"deprecated.html":[0],
"dir_02788f08a069cd285007b2a4cebef094.html":[2,0,0,0,3],
"dir_6ce568873c3b8b02e37fdfc4f2d20333.html":[2,0,0,0,2],
"dir_877e8eb39d95030c3dfad448b07343f5.html":[2,0,0,0],
"dir_9076ed57cc5e597276e884f690351ed9.html":[2,0,0,0,1],
"dir_91d1cee8e7318b3442114a0fe4379960.html":[2,0,0,0,0],
"dir_f9f156690f2dff11ed31d883dd23097f.html":[2,0,0],
"files.html":[2,0],
"functions.html":[1,3,0],
"functions_func.html":[1,3,1],
"functions_prop.html":[1,3,2],
"hierarchy.html":[1,2],
"index.html":[],
"interface_f_config.html":[1,0,0],
"interface_f_config.html#a0773bc481493a0351e2c56ee4a65d355":[1,0,0,5],
"interface_f_config.html#a516b7d845af23a5abeb6642b6f6aea13":[1,0,0,4],
"interface_f_config.html#a59867d9456a86f4dde96e3eccb5ccd1c":[1,0,0,0],
"interface_f_config.html#a95886d178981c55d956b43a46e5ea4ce":[1,0,0,3],
"interface_f_config.html#ab071dc7ff9e4fa4fbdbd358847febdd2":[1,0,0,1],
"interface_f_config.html#af95563e7446306c8ba93fd6153304d80":[1,0,0,2],
"interface_flurry.html":[1,0,2],
"interface_flurry_ad_banner.html":[1,0,3],
"interface_flurry_ad_banner.html#a2437af7b3137a2b8dc74971b45442ee5":[1,0,3,3],
"interface_flurry_ad_banner.html#a40eca9bc9f5faa460877e89d863643e8":[1,0,3,2],
"interface_flurry_ad_banner.html#a6ea7b83935ea721a3db6f4480d4f6cec":[1,0,3,7],
"interface_flurry_ad_banner.html#a855875abe2729b15446c71ddbd33a6d4":[1,0,3,0],
"interface_flurry_ad_banner.html#adf59eed9ffc93a2651ae81ac87eb49ad":[1,0,3,1],
"interface_flurry_ad_banner.html#ae4b5ed28a94efcf7d40ebe145f8bf25e":[1,0,3,4],
"interface_flurry_ad_banner.html#afb79074aa07a6e339ed5f0545202ca97":[1,0,3,5],
"interface_flurry_ad_banner.html#afd94236416f04f5ef36c8f800fd2b891":[1,0,3,6],
"interface_flurry_ad_interstitial.html":[1,0,6],
"interface_flurry_ad_interstitial.html#a3548d0b0dfcc60640656db34181c0916":[1,0,6,2],
"interface_flurry_ad_interstitial.html#a4f08d77957ce984819fb05ae27edc212":[1,0,6,0],
"interface_flurry_ad_interstitial.html#a527674749cdef0c75da3aab1b8e3ba1e":[1,0,6,3],
"interface_flurry_ad_interstitial.html#a8a2d31b8390aa670ea70df637d580433":[1,0,6,4],
"interface_flurry_ad_interstitial.html#a8dff6d55f20df82c36a5c76257e0beed":[1,0,6,6],
"interface_flurry_ad_interstitial.html#aad64e5430b9acfefb23c5f0cb31ab1de":[1,0,6,1],
"interface_flurry_ad_interstitial.html#ab768a4afa7539050984dbb68b3ed6da0":[1,0,6,5],
"interface_flurry_ad_native.html":[1,0,8],
"interface_flurry_ad_native.html#a14d33a16acf647211905b1a70dcb7ba9":[1,0,8,4],
"interface_flurry_ad_native.html#a1fe0f40954a53d9f42d19baab87f4cd0":[1,0,8,9],
"interface_flurry_ad_native.html#a229c8c56ec909d3245ee37e2c1d40a67":[1,0,8,15],
"interface_flurry_ad_native.html#a320563dd406a019e4687134c07cdd6ab":[1,0,8,7],
"interface_flurry_ad_native.html#a57bce4e4583329c78202193be48e6ae9":[1,0,8,1],
"interface_flurry_ad_native.html#a591e1fcd4df9766ce6c50f3f6af326ea":[1,0,8,13],
"interface_flurry_ad_native.html#a59bfc8aab67569dee1fc9965ddab3033":[1,0,8,0],
"interface_flurry_ad_native.html#a65370a089d7814416b4511fdb0b2dd66":[1,0,8,8],
"interface_flurry_ad_native.html#a661aa88c4a33ce7220fb10326202589e":[1,0,8,2],
"interface_flurry_ad_native.html#a7336a0cde2dc7ec0f9ae1c432afec4f3":[1,0,8,14],
"interface_flurry_ad_native.html#a7446ceff00ebb1f42db3536f1ef3928c":[1,0,8,11],
"interface_flurry_ad_native.html#a7471e00bc418f4e6a724dde504cb67aa":[1,0,8,5],
"interface_flurry_ad_native.html#a89286ea9a48e2576afde64bf41436e12":[1,0,8,12],
"interface_flurry_ad_native.html#a8da67a3606163b4f4e0220878d23249e":[1,0,8,10],
"interface_flurry_ad_native.html#a942d35995fc5929bb462a1d1d79025e8":[1,0,8,16],
"interface_flurry_ad_native.html#adc51d0ea9b92aafb2a0f5c8037f4f1b0":[1,0,8,6],
"interface_flurry_ad_native.html#aeee0bc05028d7ca2d5272508294a1c92":[1,0,8,3],
"interface_flurry_ad_native_asset.html":[1,0,9],
"interface_flurry_ad_native_asset.html#a2fff1727c69d764ed0a6b7e7804a7db0":[1,0,9,3],
"interface_flurry_ad_native_asset.html#a50759eca02d4332a0737b2e9eeeba303":[1,0,9,0],
"interface_flurry_ad_native_asset.html#a885b639a4b00a479595da6c9a33d5aae":[1,0,9,2],
"interface_flurry_ad_native_asset.html#aaa3245e98b4b073578c46cd49d79eb39":[1,0,9,4],
"interface_flurry_ad_native_asset.html#af839e5a8d6a44ca316601616fa1f4f97":[1,0,9,1],
"interface_flurry_ad_native_style.html":[1,0,11],
"interface_flurry_ad_native_style.html#a5bb31f8e74e13e15c58d6ca1a3cbf3e4":[1,0,11,0],
"interface_flurry_ad_native_style.html#aa55457625a301bdbcb41e07c6069c718":[1,0,11,1],
"interface_flurry_ad_targeting.html":[1,0,14],
"interface_flurry_ad_targeting.html#a1c85fc0043d8909f1254a3d9145c7994":[1,0,14,2],
"interface_flurry_ad_targeting.html#a2600adbd7bc010f3733dabe2581ccfd8":[1,0,14,1],
"interface_flurry_ad_targeting.html#a27cce26d3aaa140baf1f8379dd62145c":[1,0,14,0],
"interface_flurry_ad_targeting.html#af9f7f2f48713ab6699394fa080c7963c":[1,0,14,3],
"interface_flurry_ads.html":[1,0,12],
"interface_flurry_image_share_parameters.html":[1,0,16],
"interface_flurry_image_share_parameters.html#a691f5e58b79630328ae6ab96c917ddb2":[1,0,16,0],
"interface_flurry_image_share_parameters.html#aa774f36e44da2e7e709297f03f5b58c0":[1,0,16,1],
"interface_flurry_session_builder.html":[1,0,17],
"interface_flurry_session_builder.html#a1c8910aefa70cc2f76cc5b21ed6c34ef":[1,0,17,5],
"interface_flurry_session_builder.html#a1dc8f27408ae2e4a95b073bffbe1af19":[1,0,17,2],
"interface_flurry_session_builder.html#a4c65e0c2424750c857eb8b2426cd6916":[1,0,17,3],
"interface_flurry_session_builder.html#a759df1999a1212d59b7153cc7524da6e":[1,0,17,0],
"interface_flurry_session_builder.html#a8bf13ca5dd0d60d56d2619982a3b06f6":[1,0,17,1],
"interface_flurry_session_builder.html#ac891dc68740af34d6aad9098981e1cd0":[1,0,17,4],
"interface_flurry_session_builder.html#ad23c59d06218aba6cd9e4700d7a3564f":[1,0,17,6],
"interface_flurry_text_share_parameters.html":[1,0,18],
"interface_flurry_text_share_parameters.html#ab2d928a8f65099a9afb7436b68cc2e8d":[1,0,18,0],
"interface_flurry_text_share_parameters.html#ae69f60a575459eb16dfab684dfd2c789":[1,0,18,1],
"interface_flurry_tumblr.html":[1,0,19],
"interface_flurry_watch.html":[1,0,21],
"pages.html":[],
"protocol_f_config_observer-p.html":[1,0,1],
"protocol_f_config_observer-p.html#a8d54acdd4ddfa5b1f4af04357b5af786":[1,0,1,0],
"protocol_f_config_observer-p.html#aaba342f374b530224b27c07dca3d7d4c":[1,0,1,2],
"protocol_f_config_observer-p.html#ab6eea84a859a838792151adb035654a7":[1,0,1,1],
"protocol_f_config_observer-p.html#aba66b068dfb66c694d32a9ead833fc94":[1,0,1,3],
"protocol_flurry_ad_banner_delegate-p.html":[1,0,4],
"protocol_flurry_ad_banner_delegate-p.html#a27719ff9204e8d3201c677300c9e9a47":[1,0,4,4],
"protocol_flurry_ad_banner_delegate-p.html#a29db921606153bbbdfc76260aa8c3942":[1,0,4,0],
"protocol_flurry_ad_banner_delegate-p.html#a50be31f2577ec1ff3a4789b23715bd29":[1,0,4,1],
"protocol_flurry_ad_banner_delegate-p.html#a513643cdc61bad82fe502a8a62feacbf":[1,0,4,5],
"protocol_flurry_ad_banner_delegate-p.html#a53aa67e44dc264173ece6c1bcd333054":[1,0,4,2],
"protocol_flurry_ad_banner_delegate-p.html#a606f03026e78a90c82442e4e893bf8d5":[1,0,4,3],
"protocol_flurry_ad_banner_delegate-p.html#a639722aced118d82e5fe670b51159c38":[1,0,4,8],
"protocol_flurry_ad_banner_delegate-p.html#ae7c2c86e4cd9869fb97cef78484a33e0":[1,0,4,6],
"protocol_flurry_ad_banner_delegate-p.html#aebc97c95931c473a78581b774520ba3d":[1,0,4,7],
"protocol_flurry_ad_delegate-p.html":[1,0,5],
"protocol_flurry_ad_delegate-p.html#a0e587dc73e939ddeb1fafd599c5100a6":[1,0,5,19],
"protocol_flurry_ad_delegate-p.html#a1c9789a9698d15ad69c9f4a6d007995b":[1,0,5,6],
"protocol_flurry_ad_delegate-p.html#a221193d2ed566d603b1fec34161afeca":[1,0,5,28],
"protocol_flurry_ad_delegate-p.html#a28f2083dcdad2cfdf6bc092f0b3920d3":[1,0,5,4],
"protocol_flurry_ad_delegate-p.html#a320875211530c69c2c4fc99f0f0b729f":[1,0,5,9],
"protocol_flurry_ad_delegate-p.html#a4556ec4b408f3ebb3a0fd5e558da66a8":[1,0,5,20],
"protocol_flurry_ad_delegate-p.html#a5106469035dab1103e0f8e7815d8d23e":[1,0,5,16],
"protocol_flurry_ad_delegate-p.html#a697ef9711a40d6bc3a175f1f27dca051":[1,0,5,22],
"protocol_flurry_ad_delegate-p.html#a6ab7ef0ce17c2b1c9f99dd34d1a35545":[1,0,5,8],
"protocol_flurry_ad_delegate-p.html#a8c8a073785f5352c5de473a8ac8f0f81":[1,0,5,10],
"protocol_flurry_ad_delegate-p.html#a8f1c3c434dd276551d4fb1357be268b6":[1,0,5,23],
"protocol_flurry_ad_delegate-p.html#a957280502a3ecce580a35e756937cdd1":[1,0,5,2],
"protocol_flurry_ad_delegate-p.html#a96b95c262bbaaa638465c26d89e14fbf":[1,0,5,7],
"protocol_flurry_ad_delegate-p.html#aa5b80ed4df4122d245c19684954432e7":[1,0,5,3],
"protocol_flurry_ad_delegate-p.html#aa72eff6703fb3abf9054148feaebc291":[1,0,5,12],
"protocol_flurry_ad_delegate-p.html#aac831836685267b41b9bbc00ca392842":[1,0,5,15],
"protocol_flurry_ad_delegate-p.html#aacd4b0c27a568e42f0115df3e333d7e6":[1,0,5,5],
"protocol_flurry_ad_delegate-p.html#ab10225490cbe9a79bbb716291100e85d":[1,0,5,25],
"protocol_flurry_ad_delegate-p.html#ab1c2a4bc194dabf7ce973efa0e42beff":[1,0,5,1],
"protocol_flurry_ad_delegate-p.html#ab3aab75fc98595df6dfff08cb4762c76":[1,0,5,18],
"protocol_flurry_ad_delegate-p.html#ab48c7bbba044873d95f894f84d2a128a":[1,0,5,13],
"protocol_flurry_ad_delegate-p.html#ab7aacea36c78671b9feacd0dd0681206":[1,0,5,27],
"protocol_flurry_ad_delegate-p.html#acded2c10dd666b403851e9d5aa732e13":[1,0,5,11],
"protocol_flurry_ad_delegate-p.html#ae110ac801011a97400b31643fede7007":[1,0,5,24],
"protocol_flurry_ad_delegate-p.html#ae8a35856ece8780b0f241ecedf1a4047":[1,0,5,21],
"protocol_flurry_ad_delegate-p.html#af3d0fd7e16ead501f8c8f93c74651a81":[1,0,5,17],
"protocol_flurry_ad_delegate-p.html#af9eda4962cc12f7e4aeb197e9eb943ea":[1,0,5,0],
"protocol_flurry_ad_delegate-p.html#aff0b7c92964bdb8568985c61cad696b3":[1,0,5,26],
"protocol_flurry_ad_delegate-p.html#aff43a6fbd2a1d34170d25721c013bf9f":[1,0,5,14],
"protocol_flurry_ad_interstitial_delegate-p.html":[1,0,7],
"protocol_flurry_ad_interstitial_delegate-p.html#a06dffa200d20497eea9b3b8c58ff5093":[1,0,7,2],
"protocol_flurry_ad_interstitial_delegate-p.html#a1dcf83e2e6aa2e3589dd617a6ad49a39":[1,0,7,7],
"protocol_flurry_ad_interstitial_delegate-p.html#a2d5cfd5f1f71d2e1b4bff67da57de9d1":[1,0,7,4],
"protocol_flurry_ad_interstitial_delegate-p.html#a3446e0d7dcafb939fba5080c5e700533":[1,0,7,8],
"protocol_flurry_ad_interstitial_delegate-p.html#a8ca9e3d0a07aa8fd069062eeb2fe2903":[1,0,7,1],
"protocol_flurry_ad_interstitial_delegate-p.html#aa487cb1fe1326d04189d8c729ef7fa69":[1,0,7,0],
"protocol_flurry_ad_interstitial_delegate-p.html#ac3cb90aa5f4fc11cd99aff818e2dee75":[1,0,7,3],
"protocol_flurry_ad_interstitial_delegate-p.html#ade8391cc2190b453a8470da57031de4a":[1,0,7,5],
"protocol_flurry_ad_interstitial_delegate-p.html#aeee375d209dba215b575f4390c583869":[1,0,7,6],
"protocol_flurry_ad_native_delegate-p.html":[1,0,10],
"protocol_flurry_ad_native_delegate-p.html#a145a00a1cfd5526c157bdbcf7af7a4a8":[1,0,10,2],
"protocol_flurry_ad_native_delegate-p.html#a257a0d54437bfffdb184e7a1464ad7cd":[1,0,10,7],
"protocol_flurry_ad_native_delegate-p.html#a27cbc783d56a333c7639e96f8aba5637":[1,0,10,5],
"protocol_flurry_ad_native_delegate-p.html#a2f5f3d7b688c95240f51140c848404d3":[1,0,10,6],
"protocol_flurry_ad_native_delegate-p.html#a65b451548e89f3bd16e48ce8dfcb9363":[1,0,10,8],
"protocol_flurry_ad_native_delegate-p.html#a74707f0902106972554b42a36495733b":[1,0,10,4],
"protocol_flurry_ad_native_delegate-p.html#a8815fca110a389d44754ae3b995f2c97":[1,0,10,0],
"protocol_flurry_ad_native_delegate-p.html#a9fd3605a6413e3918296e2ec5db418fc":[1,0,10,3],
"protocol_flurry_ad_native_delegate-p.html#aef9b695e48670156d1c9ca9030e0fa8b":[1,0,10,1],
"protocol_flurry_ad_stream_delegate-p.html":[1,0,13],
"protocol_flurry_ad_stream_delegate-p.html#a1bc0052333f165606b352ac5791af41e":[1,0,13,6],
"protocol_flurry_ad_stream_delegate-p.html#a61043f3dbfa9cdcf0b0a459b1c604528":[1,0,13,9],
"protocol_flurry_ad_stream_delegate-p.html#a730f1fde7b03640959c5ff5a93ac53da":[1,0,13,8],
"protocol_flurry_ad_stream_delegate-p.html#a9a81ae5d4439698d24e0536d9f7797de":[1,0,13,7],
"protocol_flurry_ad_stream_delegate-p.html#aa36396bd97171c0dc329365d4a41fa24":[1,0,13,2],
"protocol_flurry_ad_stream_delegate-p.html#aa7be10d99916e82ee704fca0408abfa2":[1,0,13,1],
"protocol_flurry_ad_stream_delegate-p.html#ab726067c0b2b7c22c34cd7e8cdca526b":[1,0,13,5],
"protocol_flurry_ad_stream_delegate-p.html#abaac81503d6aeac37449d4897bc17636":[1,0,13,0],
"protocol_flurry_ad_stream_delegate-p.html#adde4a99d78ab5eb24a0f44abe1474bbf":[1,0,13,3],
"protocol_flurry_ad_stream_delegate-p.html#ae544d7ec336deec375dda8b08bf32ef9":[1,0,13,4],
"protocol_flurry_delegate-p.html":[1,0,15],
"protocol_flurry_delegate-p.html#a6dc78e394f0aa01cdcca997525e07fe4":[1,0,15,0],
"protocol_flurry_tumblr_delegate-p.html":[1,0,20],
"protocol_flurry_tumblr_delegate-p.html#a84b77dd27276943f3dc444fa6abe8033":[1,0,20,1],
"protocol_flurry_tumblr_delegate-p.html#a8b779f8966af8e67e052e57daace9aa0":[1,0,20,0],
"protocol_i_flurry_tumblr_share_parameters-p.html":[1,0,22],
"protocol_i_flurry_tumblr_share_parameters-p.html#a4db804a82877458a480cfdf0ddd1a4a0":[1,0,22,1],
"protocol_i_flurry_tumblr_share_parameters-p.html#acd4f1247c5dea8294ec615d1ca331fbf":[1,0,22,0],
"protocol_i_flurry_tumblr_share_parameters-p.html#af4b4b585118a74e51b6dcb43058cb923":[1,0,22,2]
};
